sklearn/utils/tests/test_utils.py

```diff
@@ -657,9 +657,9 @@ def test_print_elapsed_time(message, expected, capsys, monkeypatch):
 
 @pytest.mark.parametrize("value, result", [(float("nan"), True),
                                            (np.nan, True),
-                                           (np.float("nan"), True),
-                                           (np.float32("nan"), True),
-                                           (np.float64("nan"), True),
+                                           (np.float(float("nan")), True),
+                                           (np.float32(float("nan")), True),
```

Comment:
The signature of these functions is,
```
class float64(floating, builtins.float)
```
it does seem to work with strings, probably because `float` is called on them internally, but technically that's not correct.